ftdetect/bzl.vim

```diff
@@ -12,4 +12,7 @@
 " See the License for the specific language governing permissions and
 " limitations under the License.
 
-autocmd BufRead,BufNewFile *.bzl,BUILD,*.BUILD,BUILD.*,WORKSPACE setfiletype bzl
+autocmd BufRead,BufNewFile *.bzl,WORKSPACE setfiletype bzl
+if has('fname_case')
+  autocmd BufRead,BufNewFile BUILD,*.BUILD,BUILD.* setfiletype bzl
```

Comment:
The poster on the mailing list says they're using OS X.

Also Vim sets `CASE_INSENSITIVE_FILENAME` on OS X: https://github.com/vim/vim/blob/v8.0.0563/src/os_mac.h#L91